### ISSUE TITLE
Update error "has additional properties" to have array with additional properties names

### ISSUE THREAD

**Issue by ceoworks** [mafintosh/is-my-json-valid#102]
Suppose we have a schema with such a properties `userSchema = {name, surname}`
But in he input we receive object that looks like `input = {name, surname, login}`
When validate accross userSchema, error must contain 'forbiddenProperties' array besides message and property fields.
Example: `{ message: 'has additional properties', property: 'data', additonalProperties: [ 'login' ] }`

Kind Regards

**Comment by LinusU** [comment]
I think that this would be awesome!

**Comment by dmeznaric** [comment]
this is already possible. You only need to define schema properly.
For your example @ceoworks login should be defined as optional property.

Something like this:
```javascript
 var schema = {
        type: 'object',
        properties: {
          name: {type: 'string' },
          surname: {type: 'string' },
          login: {type: 'string' }
        },
        required: [ "name", "surname" ],
        optional: [ "login" ],
        additionalProperties: false
      };
```

**Comment by emilbayes** [comment]
I believe this can be closed given @dmeznaric solution

**Comment by LinusU** [comment]
I don't think that this is resolved, the original issue was about giving the property that was extra _in the error output._ I don't think that this is solved yet...
